includes/Customizer.php

```diff
@@ -0,0 +1,593 @@
+<?php
+/**
+ * Class Customizer
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Class customizer settings.
+ *
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ *
+ * @package Google\Web_Stories
+ */
+class Customizer {
+
+	/**
+	 * Customizer section slug.
+	 *
+	 * @var string
+	 */
+	const SECTION_SLUG = 'web_story_options';
+
+	/**
+	 * WP_Customize_Manager instance.
+	 *
+	 * @var \WP_Customize_Manager $wp_customize WP_Customize_Manager instance.
+	 */
+	private $wp_customize;
+
+	/**
+	 * Initializes the customizer logic.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action( 'customize_register', [ $this, 'register_customizer_settings' ] );
+	}
+
+	/**
+	 * Registers web stories customizer settings.
+	 *
+	 * @param \WP_Customize_Manager $wp_customize WP_Customize_Manager instance.
+	 *
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+	 *
+	 * @return void
+	 */
+	public function register_customizer_settings( \WP_Customize_Manager $wp_customize ) {
+
+		$this->wp_customize = $wp_customize;
+
+		$theme_support = self::get_stories_theme_support();
+
+		// Add Content section.
```

Comment:
I once heard @westonruter speak about registering customizer settings. I believe he said it better to register settings in javascript, as all registering them in PHP does is dumps a json blob on the page that js then uses to register. 

Wonder if @swissspidy also has some thoughts on the matter. Selective refresh would also be really good here.

Reply:
How many settings are being registered? When there are a _lot_ of settings (e.g. widgets or nav menu items), then it's better to register them with JS lazily (such as when the Customizer panel is opened). Otherwise, registering them with JS is probably overkill. It looks like there are 15 settings here? Then using PHP is fine.

Reply:
@westonruter Thank you for responding.
To be precise, there are 12 settings over here, so it should be fine with PHP.